virttest/utils_time.py

```diff
@@ -76,31 +76,24 @@ def verify_timezone_win(session):
     """
     def get_timezone_list():
         timezone_list_cmd = "tzutil /l"
-        timezone_set = []
-        timezone_sets = []
         timezone_list = session.cmd_output_safe(timezone_list_cmd)
+        match_pattern = "(?:\(UTC([+|-]\d{2}:\d{2})?)(?:.*\n)(\w+.*(?:\s\w+)*)"
 
-        for line in timezone_list.splitlines():
-            # Empty line
-            if not line:
-                continue
-            _code = re.search(r'(?:\(UTC([+|-]\d{2}:\d{2})?)', line)
-            if _code and _code.group(1):
-                _code_ = re.sub(r'(\d{2}):(\d{2})', r'\1\2',
-                                _code.group(1))
-                timezone_set.append(_code_)
+        timezone_sets = []
+        codes, names = [], []
+        for para in re.split("(?:\r?\n){2,}", timezone_list.strip()):
+            try:
+                code, name = re.match(match_pattern, para, re.M).groups()
+            except AttributeError:
                 continue
             # When UTC standard time, add timezone code '+0000'
-            elif _code:
-                timezone_set.append("+0000")
-                continue
-            _name = re.search(r'(\S+(?:\s\S+)*)', line)
-            if _name:
-                timezone_set.append(_name.group(0))
+            if not code:
+                code = "+0000"
             else:
-                logging.warn("Can not get timezone name correctly.")
-            timezone_sets.append(timezone_set)
-            timezone_set = []
+                code = re.sub(r'(\d{2}):(\d{2})', r'\1\2', code)
+            codes.append(code)
+            names.append(name)
+        timezone_sets = list(zip(codes, names))
```

Comment:
Suggest:
```python
match_pattern = "(?:\(UTC([+|-]\d{2}:\d{2})?)(?:.*\n)(\w+.*(?:\s\w+)*)"
timezone_sets = []
for para in re.split("(?:\r?\n){2,}", timezone_list.strip()):
    result = re.match(match_pattern, para, re.M).groups()
    if not result:
        continue
    code, name = result.groups()
    # When UTC standard time, add timezone code '+0000'
    if not code:
        code = "+0000"
    else:
        code = re.sub(r'(\d{2}):(\d{2})', r'\1\2', code)
    timezone_sets.append((code, name))
```

Reply:
Thanks, updated code, please help review it again, thanks.

Test pass.
 (1/1) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.timedrift.shared_ntp_date.with_reboot: PASS (349.02 s)